components/glew/Makefile

```diff
@@ -0,0 +1,100 @@
+#
+# This file and its contents are supplied under the terms of the
+# Common Development and Distribution License ("CDDL"). You may
+# only use this file in accordance with the terms of the CDDL.
+#
+# A full copy of the text of the CDDL should have accompanied this
+# source. A copy of the CDDL is also available via the Internet at
+# http://www.illumos.org/license/CDDL.
+#
+
+#
+# Copyright 2016 Jim Klimov <jimklimov@gmail.com>
+#
+
+# NOTE: This project links with /usr/openwin/lib/libGL.so.1, which unrolls
+# to /usr/lib/GL/libGL.so.1 which is a symlink to run-time (hardware-related?)
+# mediator in the ../../../var/run/opengl/lib/libGL.so.1. So far so good...
+# In my case (GZ on a VirtualBox host, even though it is dual-booted and runs
+# OI/Hipster natively at the moment) this library symlink points back to
+# /usr/lib/mesa/libGL.so.1 and this one object points to /usr/lib/VBoxOGL.so
+# rather than the nearby mesa objects. This breaks the build because pkgmogrify
+# insists on objects coming from known repo's, and because ultimate objects do
+# refer to virtualbox rather than libs expected available on all deployments!
+# Perhaps we should *require* do the builds in a dedicated local zone?..
+# TODO: Find out the root-cause and attach a patch-file with a fix to this.
+# So far we have a check to abort the build if this condition is detected.
+
+include ../../make-rules/shared-macros.mk
+
+COMPONENT_NAME=		glew
+COMPONENT_VERSION=	1.13.0
+COMPONENT_SUMMARY=	The OpenGL Extension Wrangler Library
+COMPONENT_CLASSIFICATION=org.opensolaris.category.2008:System/Libraries
```

Comment:
Well, at the very least it makes sense to claim some one package fmri for
both versions; i can adapt in my PR;)

Also, why is theirs encumbered?

And no MACH64 for libs is a bad thing. We might pick what bins to build,
but libs should be available in both arch's imho. Otherwise it blocks
future evolution into thre absent arch. Been there ;)


вторник, 9 февраля 2016 г. пользователь Aurélien Larcher написал:
> In components/glew/Makefile:
>
>> +# In my case (GZ on a VirtualBox host, even though it is dual-booted
and runs
>> +# OI/Hipster natively at the moment) this library symlink points back to
>> +# /usr/lib/mesa/libGL.so.1 and this one object points to
/usr/lib/VBoxOGL.so
>> +# rather than the nearby mesa objects. This breaks the build because
pkgmogrify
>> +# insists on objects coming from known repo's, and because ultimate
objects do
>> +# refer to virtualbox rather than libs expected available on all
deployments!
>> +# Perhaps we should *require* do the builds in a dedicated local zone?..
>> +# TODO: Find out the root-cause and attach a patch-file with a fix to
this.
>> +# So far we have a check to abort the build if this condition is
detected.
>> +
>> +include ../../make-rules/shared-macros.mk
>> +
>> +COMPONENT_NAME= glew
>> +COMPONENT_VERSION= 1.13.0
>> +COMPONENT_SUMMARY= The OpenGL Extension Wrangler Library
>> +COMPONENT_CLASSIFICATION=org.opensolaris.category.2008:System/Libraries
>
> Phorcys' branch is actually a contribution that we are reviewing to get
Kodi in oi-userland.
> I thought this was a good opportunity to also merge his changes in the
process.
>
> —
> Reply to this email directly or view it on GitHub.<
https://ci4.googleusercontent.com/proxy/rxLDwdgJV3OIuAX4SN6hMAGgQoiveCSxgCLxaiJAvNgdA3CETG6CWzcWGEmuIqRsDOTUAuGEygQxm_7b4ePdKv4F05pCZKv_HwcJnbz5FxOSvz5V4jBWPSA4CdY83GNnyTvj1ZQE2C8vUS0cRlE51L-ctp005g=s0-d-e1-ft#https://github.com/notifications/beacon/ABj5lKKjj-uJFaVrQNGuJ-nctvAs0eKVks5piTgjgaJpZM4HVpt7.gif
>